plugins/chromatic_correction.py

```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+"""
+Created on June 13, 2024
+
+@author: Karishma Kumar
+
+Gives ability to provide charomatic correction in x and y based on the given streams in the
+localisation tab.
+
+Copyright © 2024 Karishma Kumar, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not,
+see http://www.gnu.org/licenses/.
+"""
+
+import copy
+
+import cv2
+import numpy
+import wx
+import yaml
+from scipy.spatial.distance import cdist
+
+from odemis import model, dataio
+from odemis.acq.acqmng import acquireZStack
+from odemis.acq.stream import FluoStream
+from odemis.gui.plugin import Plugin
+from odemis.util.comp import generate_zlevels
+from odemis.util.transform import AffineTransform
+
+BEAD_DIAMETER = 10 * numpy.sqrt(2)  # the length of the square that fits in the bead circle
+
+
+def process_image(image):
+    # Check the number of channels in the image
+    if len(image.shape) == 2:  # Grayscale image
+        gray = image
+    elif len(image.shape) == 3 and image.shape[2] == 3:  # BGR image
+        gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    else:
+        raise ValueError("Unexpected image format!")
+
+    # Apply threshold
+    _, binary = cv2.threshold(gray, 200, 255, cv2.THRESH_BINARY)
```

Comment:
It is based on the dataset provided on chromatic aberration. Tested their on the images. I can provide a formulation for this number based on the histogram, to avoid fixing it.

Reply:
I am not able to find any formulation for this magic number, I tried what otsu thresholding provides and it does not match with this number. Thresholding with this number worked on the images provided by Deniz to calculate lateral chromatic shift such that after thresholding it keeps the size of beads is same in binary image as in gray scale image. When Otsu thresholding is used, the beads in binary image are dilated. The dilation would affect the operations using bead diameter. If it is fine, I can make it a global parameter  or a input parameter for the functions and use it like this for now. As it is tested more, we can find a better threshold if it does not work ?